### PR TITLE
Remove deprecated QDesktopWidget

### PR DESCRIPTION
**Pull request by luis-pereira**
It maintains functionality. It uses the default/primary screen.

### CODE REVIEW

File: lxqtpowermanager.cpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ class LXQT_API MessageBox: public QMessageBox
     void resizeEvent(QResizeEvent* event) override
     {
         Q_UNUSED(event)
-        const QRect screen = QApplication::desktop()->screenGeometry();
+        const QRect screen = QGuiApplication::primaryScreen()->geometry();
```
Comment (tsujan) [inline lxqtpowermanager.cpp]: @luis-pereira 
To be exact, according to Qt → `qdesktopwidget.cpp`, it is so:
```c++
        auto screens = QApplication::screens();
        const QRect screen = screens.isEmpty() ? QRect() : screens.at(0)->geometry();
```

Comment (tsujan) [reply lxqtpowermanager.cpp]: Explanation:

In `qdesktopwidget.cpp`, the method `QDesktopWidgetPrivate::screen(int screenNo)` is used with 1 for `screenNo` in `QDesktopWidgetPrivate::screenGeometry(int screenNo)`:

```c++
QScreen *QDesktopWidgetPrivate::screen(int screenNo)
{
    QList<QScreen *> screens = QGuiApplication::screens();
    if (screenNo == -1)
        screenNo = 0;
    if (screenNo < 0 || screenNo >= screens.size())
        return nullptr;
    return screens.at(screenNo);
}
```

Comment (luis-pereira) [reply lxqtpowermanager.cpp]: @tsujan 
QGuiApplication::primaryScreen() already returns the  screen.at(0). But it also can return a null pointer. My code doesn't handles that.

A better solution is:
``` diff
-        const QRect screen = QApplication::desktop()->screenGeometry();
+        QScreen *primaryScreen = QGuiApplication::primaryScreen();
+        const QRect screen = primaryScreen ? primaryScreen->geometry() : QRect();
```

Comment (tsujan) [reply lxqtpowermanager.cpp]: > QGuiApplication::primaryScreen() already returns the screen.at(0)

You're right.

> A better solution is:

They're completely equivalent to each other.

Comment (luis-pereira) [reply lxqtpowermanager.cpp]: @tsujan I'm traveling and occupying the wait times with some boring tasks. I've written my comment without reading your second comment.

Comment (tsujan) [reply lxqtpowermanager.cpp]: >  I've written my comment without reading your second comment.

And your comment was correct :) Add a nullity check and merge it.

Safe journey!

Comment (luis-pereira) [reply lxqtpowermanager.cpp]: > > 
> 
> And your comment was correct :) Add a nullity check and merge it.

``` cpp
const QRect screen = primaryScreen ? primaryScreen->geometry() : QRect();
```
already does it.
> 
> Safe journey!

Thanks

**Comment by luis-pereira** [comment]
Updated.

**Pull request merged**